lib/endpoints/class-wp-rest-terms-controller.php

```diff
@@ -528,6 +489,52 @@ public function delete_item( $request ) {
 	}
 
 	/**
+	 * Prepare a single term for create or update
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return object $prepared_term Term object.
+	 */
+	public function prepare_item_for_database( $request ) {
+		$prepared_term = new stdClass;
+		$prepared_term->args = array();
+
+		if ( isset( $request['name'] ) ) {
+			$prepared_term->name = $request['name'];
+			$prepared_term->args['name'] = $request['name'];
+		}
+
+		if ( isset( $request['description'] ) ) {
+			$prepared_term->args['description'] = $request['description'];
+		}
+
+		if ( isset( $request['slug'] ) ) {
+			$prepared_term->args['slug'] = $request['slug'];
+		}
+
+		if ( isset( $request['parent'] ) ) {
+			if ( ! is_taxonomy_hierarchical( $this->taxonomy ) ) {
+				return new WP_Error( 'rest_taxonomy_not_hierarchical', __( 'Can not set resource parent, taxonomy is not hierarchical.' ), array( 'status' => 400 ) );
```

Comment:
Would silently ignoring the `parent` parma be setting it to `-1` instead of throwing an error, or not setting it at all within the $prepared_term object?

Also, if I remove the returned WP_Error objects then should I remove these checks as well?

- https://github.com/WP-API/WP-API/pull/2377/files#diff-6a5f19de0f37a384ee52877de312ee67R409
- https://github.com/WP-API/WP-API/pull/2377/files#diff-6a5f19de0f37a384ee52877de312ee67R337